aws/resource_aws_elasticsearch_domain_test.go

```diff
@@ -13,6 +15,55 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_elasticsearch_domain", &resource.Sweeper{
+		Name: "aws_elasticsearch_domain",
+		F:    testSweepElasticSearchDomains,
+	})
+}
+
+func testSweepElasticSearchDomains(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).esconn
+
+	prefixes := []string{
+		"tf-test-",
+		"tf-acc-test-",
+	}
+
+	out, err := conn.ListDomainNames(&elasticsearch.ListDomainNamesInput{})
+	if err != nil {
+		return fmt.Errorf("Error retrieving Elasticsearch Domains: %s", err)
+	}
+	for _, domain := range out.DomainNames {
+		skip := false
+		for _, prefix := range prefixes {
+			if !strings.HasPrefix(*domain.DomainName, prefix) {
+				skip = true
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping Elasticsearch Domain: %s", *domain.DomainName)
+			continue
+		}
+		log.Printf("[INFO] Deleting Elasticsearch Domain: %s", *domain.DomainName)
+
+		_, err := conn.DeleteElasticsearchDomain(&elasticsearch.DeleteElasticsearchDomainInput{
+			DomainName: domain.DomainName,
+		})
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete Elasticsearch Domain %s: %s", *domain.DomainName, err)
+			continue
+		}
```

Comment:
I think the convention, even for sweepers, is to wait until the resource _actually_ gets deleted, i.e. do you mind decoupling [the waiter](https://github.com/terraform-providers/terraform-provider-aws/blob/3040b33e70bf36fd184664d1d4f1bab1bb14c28b/aws/resource_aws_elasticsearch_domain.go#L689-L714) and using it here?

If we don't do that then we might still run into limits as tests will run when domains are being deleted, yet still consuming the capacity and also the subsequent run of sweepers (post-sweepers) will try to delete those domains (again).

Reply:
No problem, I'll add that in